docs/java-rest/high-level/migration/upgrade.asciidoc

```diff
@@ -0,0 +1,79 @@
+[[java-rest-high-migration-upgrade]]
```

Comment:
there is a lot of duplication between this and for instance bulk api doc. Probably some refactoring to remove doc duplication would be welcome

Reply:
yea i think that we can wait till docs squares away what they are doing with our rest api docs before diving in to this.